internals/templates/containers/App/index.js

```diff
@@ -12,18 +12,26 @@
  */
 
 import React from 'react';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
 
-export default class App extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
+import Routes from 'routes';
 
-  static propTypes = {
-    children: React.PropTypes.node,
-  };
+import { makeSelectLocation } from './selectors';
+
+export class App extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
 
   render() {
     return (
       <div>
-        {React.Children.toArray(this.props.children)}
+        <Routes />
       </div>
     );
   }
 }
+
+const mapStateToProps = createStructuredSelector({
+  location: makeSelectLocation(),
```

Comment:
Without having location in the props, App will not rerender on location change if the component is pure or connected, as react-router-v4 relies on normal react rerendering semantics. Even if location isn't used, I think it's a good default to pass location to App to minimize rerender issues.